src/main.c

```diff
@@ -36,34 +36,33 @@ typedef struct {
   struct sockaddr_storage raddr;
 } request_t;
 
-// Very very basic hostname parsing.
-// Note: Performs basic checks to see if last digit is non-alpha.
-// Non-alpha hostnames are assumed to be IP addresses. e.g. foo.1
-// Returns non-zero on success, zero on failure.
-static int hostname_from_uri(const char* uri,
-                             char* hostname, int hostname_len) {
-  if (strncmp(uri, "https://", 8) != 0) { return 0; }  // not https://
-  uri += 8;
-  const char *end = uri;
-  while (*end && *end != '/') { end++; }
-  if (end - uri >= hostname_len) {
-    return 0;
-  }
-  if (end == uri) { return 0; }  // empty string.
-  if (!isalpha(*(end - 1))) { return 0; }  // last digit non-alpha.
-
-  // If using basic authentication in URL, chop off prefix.
-  char *tmp = strchr(uri, '@');
-  if (tmp) {
-    tmp++;
-    if (tmp < end) {
-      uri = tmp;
+static int is_ipv4_address(char *str) {
+    struct in6_addr addr;
+    return inet_pton(AF_INET, str, &addr) == 1;
+}
+
+static int hostname_from_url(const char* url_in,
```

Comment:
Nit: May be worth keeping the return code comment (non-zero == success, zero = failure).

Reply:
Actually I have left a very minimal trace of that comment by indicating which value means success. I thought it is enough.
`res = 1; // success`